src/core/tree_methods.test.ts

```diff
@@ -0,0 +1,77 @@
+import * as tree from './tree_methods'
+import * as _ from 'lodash'
```

Comment:
Can we move the prototype pattern implementation insde the tree? Since it's a class we might as well use the abstraction.

Reply:
If you suggest moving lodash usage to tree_methods.ts, it is done.